app/controllers/rooms.js

```diff
@@ -196,12 +204,42 @@ module.exports = function() {
                     return res.sendStatus(400);
                 }
 
-                var user = req.user.toJSON();
-                user.room = room._id;
+                var password = req.data.password;
+                if(!!room.password && !password) {

```

Comment:
Today I learned that people do this to cast the value to a Boolean type. I'm not sure why it's needed.